### PR TITLE
Send confirmation popup

### LINKED ISSUE #3116: Add confirmation popup after send
and show the actual, total, transaction fee.

@Kirvx we are finally moving forward with this, it will be optional (it can be disabled in settings), and it will show the total transaction fee, before signing.

### PR DESCRIPTION
**Pull request by cmgustavo**
Fixes #3116

### CODE REVIEW

**Comment by cmgustavo** [comment]
Updated and ready for review

File: src/js/services/txFormatService.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ angular.module('copayApp.services').factory('txFormatService', function(profileS
 
     var outputs = lodash.isArray(tx.outputs) ? tx.outputs.length : 0;
```
Comment (matiu) [inline src/js/services/txFormatService.js]: shouldn't be outputs = 1 if no tx.outsputs defined?

File: src/js/services/txFormatService.js [hunk 2]
```diff
@@ -37,6 +37,7 @@ angular.module('copayApp.services').factory('txFormatService', function(profileS
       }, 0);
     }
 
+    tx.toAddress = tx.outputs[0].toAddress; // Get first address for single transactions
```
Comment (matiu) [inline src/js/services/txFormatService.js]: what about all txs? is tx.outputs[0] will always be defined?

File: src/js/services/txSignService.js [hunk 3]
```diff
@@ -85,10 +85,47 @@ angular.module('copayApp.services').factory('txSignService', function($rootScope
         };
 
         return cb();
+
+      });
+    });
+  }; 
+
+  root.createTx = function(opts, cb) {
```
Comment (matiu) [inline src/js/services/txSignService.js]: why txSignService includes createTx? and publishTx?

Comment (matiu) [inline src/js/services/txSignService.js]: Maybe rename it to txService?

File: src/js/controllers/index.js [hunk 4]
```diff
@@ -1453,6 +1451,19 @@ angular.module('copayApp.controllers').controller('indexController', function($r
     self.setTab(tab, reset);
   });
 
+  $rootScope.$on('Local/NeedConfirmation', function(event, txp, cb) {
```
Comment (matiu) [inline src/js/controllers/index.js]: should be 'needsConfirmation'

File: src/js/services/txFormatService.js (hunk 1 above)
Comment (cmgustavo) [inline src/js/services/txFormatService.js]: It 's not necessary. The code outside the condition will complete the required info (toAddress, amountStr, AlternativeAmountStr, feeStr).

File: src/js/services/txFormatService.js (hunk 2 above)
Comment (cmgustavo) [inline src/js/services/txFormatService.js]: You are right. It 's not always defined for old tx proposals. Fixing it right now.

**Comment by cmgustavo** [comment]
Updated. Thanks for reviewing

**Comment by matiu** [comment]
Code ACK, tested on browser.

**Pull request merged**